Loop/Managers/Alerts/AlertStore.swift

```diff
@@ -292,46 +304,41 @@ extension AlertStore {
 
 // MARK: Query Support
 
-public protocol QueryFilter: Equatable {
+public protocol QueryFilter {
     var predicate: NSPredicate? { get }
 }
 
 extension AlertStore {
 
-    public struct QueryAnchor<Filter: QueryFilter>: RawRepresentable, Equatable {
+    public struct QueryAnchor: RawRepresentable, Equatable {
```

Comment:
Why did you get rid of the `filter` in the Anchor?  Isn't it a fundamental part of the query?

Reply:
No, not really. 

1) The call to `executeQuery(since date: Date, ...` is only called for the first page, so the filter isn't needed agin.
2) The call to `executeQuery(fromQueryAnchor queryAnchor: QueryAnchor?, ...` doesn't need it.

The primary gotcha is that the query anchor from the second call is serialized and persisted, yet the filter was not part of that. Seemed like a big headache to serialize the filter appropriate so the code was consistent when the filter isn't used past the first call.